crates/sui-core/src/authority/authority_store_pruner.rs

```diff
@@ -160,25 +60,41 @@ impl AuthorityStorePruner {
         Ok(pruned)
     }
 
-    fn process_checkpoints(
+    fn prune_objects(
         perpetual_db: &Arc<AuthorityPerpetualTables>,
         checkpoint_store: &Arc<CheckpointStore>,
         num_epochs_to_retain: u64,
     ) -> anyhow::Result<()> {
-        let mut pruned_seq_number = checkpoint_store
+        let mut checkpoint_number = checkpoint_store
             .get_highest_pruned_checkpoint_seq_number()?
             .unwrap_or_default();
+        let mut checkpoint_digest = CheckpointDigest::random();
         let current_epoch = checkpoint_store
             .get_highest_executed_checkpoint()?
             .map(|c| c.epoch())
             .unwrap_or_default();
+        let mut checkpoints_in_batch = 0;
+        let mut batch_effects = vec![];
+        let mut network_total_transactions = 0;
+
+        debug!(
+            "Starting object pruning. Current epoch: {}. Latest pruned checkpoint: {}",
+            current_epoch, checkpoint_number
+        );
         loop {
-            let Some(checkpoint) = checkpoint_store.get_checkpoint_by_sequence_number(pruned_seq_number + 1)?
-                else {return Ok(());};
+            let Some(checkpoint) = checkpoint_store.get_checkpoint_by_sequence_number(checkpoint_number + 1)? else {break;};
             // checkpoint's epoch is too new. Skipping for now
             if current_epoch < checkpoint.epoch() + num_epochs_to_retain {
-                return Ok(());
+                break;
+            }
+            checkpoint_number = checkpoint.sequence_number();
+            checkpoint_digest = checkpoint.digest();
+            checkpoints_in_batch += 1;
+            if network_total_transactions == checkpoint.summary.network_total_transactions {
```

Comment:
it tries to skip processing empty checkpoints to avoid any extra rocksdb IO operations
please let me know if there's a better way to verify that checkpoint is not empty